src/extension/byok/vscode-node/xAIProvider.ts

```diff
@@ -32,4 +51,69 @@ export class XAIBYOKLMProvider extends BaseOpenAICompatibleLMProvider {
 			_instantiationService,
 		);
 	}
-}
\ No newline at end of file
+
+	private parseModelVersion(modelId: string): number | undefined {
+		const match = modelId.match(/^grok-(\d+)/);
+		return match ? parseInt(match[1], 10) : undefined;
+	}
+
+	private humanizeModelId(modelId: string): string {
+		const parts = modelId.split('-').filter(p => p.length > 0);
+		return parts.map(p => {
+			if (/^\d+$/.test(p)) {
+				return p; // keep pure numbers as-is
+			}
+			return p.charAt(0).toUpperCase() + p.slice(1);
+		}).join(' ');
+	}
+
+	protected override async getAllModels(): Promise<BYOKKnownModels> {
+		try {
+			const response = await this._fetcherService.fetch(`${this._baseUrl}/language-models`, {
+				method: 'GET',
+				headers: {
+					'Authorization': `Bearer ${this._apiKey}`,
+					'Content-Type': 'application/json'
+				}
+			});
+
+			const data = await response.json() as XAIListLanguageModelsAPIResponse;
+			if (!data.models || !Array.isArray(data.models)) {
+				throw new Error('Invalid response format from xAI API');
+			}
+			this._logService.trace(`Fetched ${data.models.length} language models from xAI`);
+			const modelList: BYOKKnownModels = this._knownModels ? { ...this._knownModels } : {};
```

Comment:
Doesn't this code have the risk of a model that is in our CDN but not available in your API key shwoing up in the list? Since we always apply our CDN list? 

The logic should be we iterate through data.model and take the capabilities from the CDN if present else sensible defaults. That way xAI is the source of truth since they know best